clvm-traits/src/match_byte.rs

```diff
@@ -1,36 +1,30 @@
-use clvmr::{
-    allocator::{NodePtr, SExp},
-    Allocator,
-};
+use num_bigint::BigInt;
 
-use crate::{Error, FromClvm, Result, ToClvm};
+use crate::{ClvmDecoder, ClvmEncoder, FromClvm, FromClvmError, ToClvm, ToClvmError};
 
 #[derive(Debug, Copy, Clone)]
 pub struct MatchByte<const BYTE: u8>;
 
-impl<const BYTE: u8> ToClvm for MatchByte<BYTE> {
-    fn to_clvm(&self, a: &mut Allocator) -> Result<NodePtr> {
-        match BYTE {
-            0 => Ok(a.null()),
-            1 => Ok(a.one()),
```

Comment:
losing this is a little bit unfortunate as well. but we can always extend `ClvmEncoder` if it's a problem

Reply:
Yeah or the Allocator Encoder implementation can check these values and perform this optimization there

Reply:
good point